python/lsst/meas/deblender/sourceDeblendTask.py

```diff
@@ -364,6 +364,15 @@ def deblend(self, exposure, srcs, psf):
             center = fp.getCentroid()
             psf_fwhm = self._getPsfFwhm(psf, center)
 
+            if not (psf_fwhm > 0):
+                if self.config.catchFailures:
+                    self.log.warning("Unable to deblend source %d: because PSF FWHM=%f is invalid.",
+                                     src.getId(), psf_fwhm)
+                    src.set(self.deblendFailedKey, True)
+                    continue
+                else:
+                    raise ValueError(f"PSF at {center} has an invalid FWHM value of {psf_fwhm}")
+
             self.log.trace('Parent %i: deblending %i peaks', int(src.getId()), len(pks))
 
             self.preSingleDeblendHook(exposure, srcs, i, fp, psf, psf_fwhm, sigma1)
```

Comment:
I don't know what this is meant for, but I wonder if this should be the place to catch this and throw an error? But I'm not sure how to continue onto the next source if `self.cofig.catchFailures` is True.

Reply:
I'm not sure what that was meant for either. I guess at some point in the distant past someone thought that it would be used for something that was never implemented.